sys/include/net/credman.h

```diff
@@ -0,0 +1,192 @@
+    ecdsa_public_key_t *client_keys;    /**< Array of clients public keys */
+    size_t client_keys_size;            /**< Size of clients_keys */
+} ecdsa_params_t;
+
+/**
+ * @brief Tag of the credential.
+ */
+typedef uint16_t credman_tag_t;
+
+/**
+ * @brief Used to identify credentials for application libraries.
+ */
+#define CREDMAN_TAG_GLOBAL  (999)
+
+/**
+ * @brief Used to signal empty/no tag
+ */
+#define CREDMAN_TAG_EMPTY   (0)
+
+/**
+ * @brief Credential types
+ */
+typedef enum {
+    CREDMAN_TYPE_EMPTY  = 0,
+    CREDMAN_TYPE_PSK    = 1,
+    CREDMAN_TYPE_ECDSA  = 2,
+} credman_type_t;
+
+/**
+ * @brief Credential informations
+ */
+typedef struct {
+    credman_type_t type;        /**< Type of the credential */
+    credman_tag_t tag;          /**< Tag of the credential */
+    union {
+        psk_params_t psk;       /**< PSK credential parameters */
+        ecdsa_params_t ecdsa;   /**< ECDSA credential parameters */
+    } params;                   /**< Credential parameters */
+} credman_credential_t;
+
+/**
+ * @brief Return values
+ */
+enum {
+    CREDMAN_OK              = 0,    /**< No error */
+    CREDMAN_EXIST           = -1,   /**< Credential already exist in system pool */
+    CREDMAN_NO_SPACE        = -2,   /**< No space in system pool for new credential */
+    CREDMAN_NOT_FOUND       = -3,   /**< Credential not found in the system pool */
+    CREDMAN_INVALID         = -4,   /**< Invalid input parameter(s) */
+    CREDMAN_TYPE_UNKNOWN    = -5,   /**< Unknown credential type */
+    CREDMAN_ERROR           = -6,   /**< Other errors */
+};
+
+/**
+ * @brief Adds a credential to the credential pool
+ *
+ * @param[in] credential    Credential to add.
+ *
+ * @return CREDMAN_OK on success
+ * @return CREDMAN_EXIST if credential of @p tag and @p type already exist
+ * @return CREDMAN_NO_SPACE if credential pool is full
+ * @return CREDMAN_TYPE_UNKNOWN if @p credential has unknown
+ *         credman_credential_t::type
+ * @return CREDMAN_INVALID if @p credential has
+ *         credman_credential_t::tag with the value of CREDMAN_TAG_EMPTY OR
+ *         credman_credential_t::type with the value of CREDMAN_TYPE_EMPTY OR
+ *         credman_credential_t::params with invalid credential parameters i.e.
+ *         the key points to NULL or has a length of 0.
+ * @return CREDMAN_ERROR on other errors
+ */
+int credman_add(const credman_credential_t *credential);
+
+/**
+ * @brief Gets a credential from credential pool
+ *
+ * @param[out] credential   Found credential
+ * @param[in] tag           Tag of credential to get
+ * @param[in] type          Type of credential to get
+ *
+ * @return CREDMAN_OK on success
+ * @return CREDMAN_NOT_FOUND if no credential with @p tag and @p type found
+ * @return CREDMAN_ERROR on other errors
+ */
+int credman_get(credman_credential_t *credential, credman_tag_t tag,
+                credman_type_t type);
+
+/**
+ * @brief Delete a credential from the system. Does nothing if
```

Comment:
```suggestion
 * @brief Delete a credential from credential pool. Does nothing if
```

At least if I interpret your comment in l.19 correctly ;-).